neqo-transport/src/crypto.rs

```diff
@@ -1253,14 +1250,14 @@ impl CryptoStates {
             cipher: TLS_AES_128_GCM_SHA256,
             next_secret: hkdf::import_key(TLS_VERSION_1_3, &[0xaa; 32]).unwrap(),
         };
-        let initials = enum_map! {
-            Version::Version1 => Some(CryptoState {
-                    tx: CryptoDxState::test_default(),
-                    rx: read(0),
+        let initials = EnumMap::from_array([
```

Comment:
Doesn't this make the code more fragile?  If we change the order in the enum (we won't, but still) then this code will still compile, but it will be wrong.

Reply:
The old code made `clippy::mem_forget` fire, and that seemed like a memory leak possibility. Maybe there is another way to avoid that?

Reply:
I went to the lengths of making the change to the enum_map code, but then I tried to test it and I don't see the error.  Isn't clippy supposed to avoid warnings like this?  I'm going to suggest it to them anyway, but when I can't reproduce it, it seems almost pointless.

Reply:
You need to turn on this warning in Cargo.toml.

Reply:
Of course 🤦.  The maintainer was receptive, so maybe we'll be able to get a revision that works for us.

Reply:
`enum-map` seems pretty dead a as a project, unfortunately: https://codeberg.org/xfix/enum-map/issues/147 I doubt we'll see another release...